### ISSUE TITLE
ClickHouse driver: an infinite loop of logs if database no longer exists

### ISSUE THREAD

**Issue by slvrtrn** [metabase/metabase#45634]
### Describe the bug

With the ClickHouse driver, if a database does not exist, the sync will trigger an infinite loop of retries and logs. It is not clear whether it is currently coming from the driver code or Metabase.<br>See [https://github.com/ClickHouse/metabase-clickhouse-driver/issues/258](<https://github.com/ClickHouse/metabase-clickhouse-driver/issues/258>)

### To Reproduce

See [https://github.com/ClickHouse/metabase-clickhouse-driver/issues/258#issuecomment-2224601648](<https://github.com/ClickHouse/metabase-clickhouse-driver/issues/258#issuecomment-2224601648>)

### Expected behavior

A single failure occurs when db does not exist, and no infinite retries that could produce an enormous amount of logs.

### Logs

*No response*

### Information about your Metabase installation

```JSON
- metabase-clickhouse-driver version: 1.50.1
- Metabase version: v0.50.6
```

### Severity

Blocking some users

### Additional context

*No response*

**Comment by paoliniluis** [comment]
@Tony-metabase you were investigating this

**Comment by Tony-metabase** [comment]
Yeah this is very related to this https://github.com/ClickHouse/metabase-clickhouse-driver/issues/242 ... is it possible that the change here https://github.com/metabase/metabase/pull/43429 caused this? 

I already see @calherries on the clickhouse issue so I believe if that is the case he is already aware

**Comment by calherries** [comment]
This is QPD, it happens on every attempt to query the DB. Here's [a slack thread](https://metaboat.slack.com/archives/C0439QDFHFZ/p1721129872287519?thread_ts=1720709864.736679&cid=C0439QDFHFZ) for more context

**Comment by crisptrutski** [comment]
I tried adding the following C3P0 options to see if it would end the retry loop, but didn't seem to make any difference:

```
   "acquireRetryAttempts"         5
   "acquireRetryDelay"            1000
   "breakAfterAcquireFailure"     true
```

**Comment by metamben** [comment]
This issue only happens with `com.mchange` (c3p0) debug logs enabled and is a documented behavior of c3p0. I think knowingly running a recursion until stack overflow is bad style, but this is what `com.mchange.v2.resourcepool.BasicResourcePool/checkoutResource` does. The fix is not using c3p0, I guess, but we're not going to do that as a fix for a debug logging issue.
